coap-gateway/service/signIn.go

```diff
@@ -258,55 +258,63 @@ func signInPostHandler(req *mux.Message, client *Client, signIn CoapSignInReq) {
 	}
 
 	if err := signIn.checkOAuthRequest(); err != nil {
-		logErrorAndCloseClient(fmt.Errorf("cannot handle sign in: %v", err), coapCodes.BadRequest)
+		logErrorAndCloseClient(fmt.Errorf("cannot handle sign in: %w", err), coapCodes.BadRequest)
 		return
 	}
 
 	jwtClaims, err := client.ValidateToken(req.Context, signIn.AccessToken)
 	if err != nil {
-		logErrorAndCloseClient(fmt.Errorf("cannot handle sign in: %v", err), coapCodes.InternalServerError)
+		logErrorAndCloseClient(fmt.Errorf("cannot handle sign in: %w", err), coapCodes.InternalServerError)
 		return
 	}
 
-	if err := jwtClaims.validateOwnerClaim(client.server.config.Clients.AuthServer.OwnerClaim, signIn.UserID); err != nil {
-		logErrorAndCloseClient(fmt.Errorf("cannot handle sign in: %v", err), coapCodes.InternalServerError)
+	err = client.server.VerifyDeviceID(client.tlsDeviceID, jwtClaims)
+	if err != nil {
+		logErrorAndCloseClient(fmt.Errorf("cannot handle sign in: %w", err), coapCodes.Unauthorized)
```

Comment:
When unauthorized is during SignIn returned, the device resets it's coapcloudconf - removes at from the storage. Correct?

Reply:
No. The Device tries to refresh the token and if the refresh token returns unauthorized then the device sets itself to failed state.